src/components/RelatedAgreementsFieldArray/RelatedAgreementField.js

```diff
@@ -19,71 +20,86 @@ import { AppIcon, Pluggable } from '@folio/stripes/core';
 import { urls } from '../utilities';
 import css from '../styles.css';
 
-export default class RelatedAgreementField extends React.Component {
-  static propTypes = {
-    id: PropTypes.string,
-    input: PropTypes.shape({
-      name: PropTypes.string,
-      value: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
-    }).isRequired,
-    meta: PropTypes.shape({
-      error: PropTypes.node,
-      touched: PropTypes.bool,
-    }).isRequired,
-    onAgreementSelected: PropTypes.func.isRequired,
-    agreement: PropTypes.shape({
-      agreementStatus: PropTypes.shape({
-        label: PropTypes.string
-      }),
-      endDate: PropTypes.string,
-      id: PropTypes.string,
-      name: PropTypes.string,
-      label: PropTypes.string,
-      startDate: PropTypes.string,
+const propTypes = {
+  agreement: PropTypes.shape({
+    agreementStatus: PropTypes.shape({
+      label: PropTypes.string
     }),
-  }
+    endDate: PropTypes.string,
+    id: PropTypes.string,
+    name: PropTypes.string,
+    label: PropTypes.string,
+    startDate: PropTypes.string,
+  }),
+  id: PropTypes.string,
+  input: PropTypes.shape({
+    name: PropTypes.string,
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+  }).isRequired,
+  meta: PropTypes.shape({
+    error: PropTypes.node,
+    touched: PropTypes.bool,
+  }).isRequired,
+  onAgreementSelected: PropTypes.func.isRequired,
+  parentAgreementId: PropTypes.string,
+  parentAgreementName: PropTypes.string,
+};
+const RelatedAgreementField = ({
+  agreement,
+  id,
+  input,
+  meta,
+  onAgreementSelected,
+  parentAgreementId,
+  parentAgreementName,
+}) => {
+  const [selfLinkedWarning, setSelfLinkedWarning] = useState(false);
 
-  static defaultProps = {
-    agreement: {},
-  }
+  let triggerButton = useRef(null);
 
-  componentDidMount() {
-    if (!get(this.props, 'input.value.id') && get(this.triggerButton, 'current')) {
-      this.triggerButton.current.focus();
+  const { change } = useForm();
+
+  useEffect(() => {
+    if (!input.value?.id && triggerButton.current) {
+      triggerButton.current.focus();
     }
-  }
+  }, [input, triggerButton]);
 
-  renderLinkAgreementButton = value => {
-    const {
-      id,
-      input: { name },
-      onAgreementSelected,
-    } = this.props;
+  useEffect(() => {
+    if (parentAgreementId === input.value?.id && !selfLinkedWarning) {
+      change(input.name, undefined);
+      setSelfLinkedWarning(true);
+    } else if (input.value && parentAgreementId !== input.value?.id && selfLinkedWarning) {
+      setSelfLinkedWarning(false);
+    }
+  }, [change, input, parentAgreementId, selfLinkedWarning, setSelfLinkedWarning]);
 
+  const renderLinkAgreementButton = value => {
+    const name = input.name;
```

Comment:
This probably isn't necessary, you can just replace name with input.name further down